### PR TITLE
Add ScriptEvent BeforeMerge, AfterMerge

### LINKED ISSUE #4380: Add two ScriptEvents, BeforeMerge, AfterMerge
**Do you want to request a *feature* or report a *bug*?**
feature
**What is the current behavior?**
No such script

**What is the expected behavior?**
When merging branches, I can perform certain programs
**Environment you encounter the issue:**
current master

**Did this work in previous version of GitExtensions (which)?**

### PR DESCRIPTION
**Pull request by chunleik**
Fixes #4380 .

Changes proposed in this pull request:
 - Add two ScriptEvents,  BeforeMerge, AfterMerge

 
Screenshots before and after (if PR changes UI):
image

What did I do to test the code and ensure quality:
 - Manual

Has been tested on (remove any that don't apply):
 - GIT 2.10 and above
 - Windows 10

### CODE REVIEW

**Comment by RussKie** [comment]
Please open an issue describing the proposal and have it referenced in the PR
Thanks

**Comment by chunleik** [comment]
Done

File: GitUI/CommandsDialogs/FormMergeBranch.cs [hunk 1]
```diff
@@ -110,7 +111,7 @@ private void OkClick(object sender, EventArgs e)
                                                                                                    addLogMessages.Checked ? (int)nbMessages.Value : (int?)null));
 
             var wasConflict = MergeConflictHandler.HandleMergeConflicts(UICommands, this, !noCommit.Checked);
-
+            ScriptManager.RunEventScripts(this, ScriptEvent.AfterMerge);
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormMergeBranch.cs]: Shouldn't this run only if it is successfully merged?

Comment (chunleik) [reply GitUI/CommandsDialogs/FormMergeBranch.cs]: You are right. This count be better.

File: GitUI/CommandsDialogs/FormMergeBranch.cs [hunk 2]
```diff
@@ -110,9 +111,10 @@ private void OkClick(object sender, EventArgs e)
                                                                                                    addLogMessages.Checked ? (int)nbMessages.Value : (int?)null));
 
             var wasConflict = MergeConflictHandler.HandleMergeConflicts(UICommands, this, !noCommit.Checked);
-
+
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormMergeBranch.cs]: please undo this

**Comment by chunleik** [comment]
OK

**Comment by RussKie** [comment]
It looks good, I'll have it tested once we are done with 2.51 release.
Until then, please stand by.

**Comment by RussKie** [comment]
Please squash and we'll get it merged
Thank you

**Pull request merged**